sensor/common/sensor/central_receiver_test.go

```diff
@@ -0,0 +1,418 @@
+	s.receiver = NewCentralReceiver(s.finished)
+	s.mockClient.EXPECT().Context().AnyTimes().Return(context.Background()).AnyTimes()
+
+	s.mockClient.EXPECT().Recv().Times(1).DoAndReturn(func() (*central.MsgToSensor, error) {
+		return nil, io.EOF
+	})
+
+	s.receiver.Start(s.mockClient)
+	s.finished.Wait()
+	s.NoError(s.receiver.Stopped().Err())
+}
+
+func (s *centralReceiverSuite) Test_SlowComponentDoesNotBlockOthers() {
+	fastTick := make(chan struct{})
+	close(fastTick)
+	fastComponent := &testSensorComponent{
+		t:          s.T(),
+		name:       "fast",
+		tick:       fastTick,
+		responsesC: make(chan *message.ExpiringMessage),
+	}
+	blockedChan := make(chan struct{})
+	blockingComponent := &testSensorComponent{
+		t:          s.T(),
+		name:       "blocked",
+		tick:       blockedChan,
+		responsesC: make(chan *message.ExpiringMessage),
+	}
+
+	components := []common.SensorComponent{fastComponent, blockingComponent}
+	s.receiver = NewCentralReceiver(s.finished, components...)
+
+	s.mockClient.EXPECT().Context().AnyTimes().Return(context.Background()).AnyTimes()
+
+	msgCount := 3
+	s.mockClient.EXPECT().Recv().MinTimes(msgCount).DoAndReturn(func() (*central.MsgToSensor, error) {
+		return testMsg, nil
+	})
+
+	s.receiver.Start(s.mockClient)
+
+	for i := 0; i < msgCount; i++ {
+		select {
+		case <-fastComponent.ResponsesC():
+			s.T().Logf("Fast component processed the message")
+		case <-blockingComponent.ResponsesC():
+			assert.FailNow(s.T(), "blocked component should not process the message")
+		}
+	}
+
+	s.receiver.Stop()
+	s.finished.Wait()
+	s.NoError(s.receiver.Stopped().Err())
+}
+
+func (s *centralReceiverSuite) Test_FilterIgnoresMessages() {
+	const (
+		numberOfCentralMessages = 5
+		queueSize               = 3
+	)
+	s.Require().NoError(os.Setenv(env.RequestsChannelBufferSize.EnvVar(), fmt.Sprintf("%d", queueSize)))
+	s.T().Cleanup(func() {
+		s.NoError(os.Unsetenv(env.RequestsChannelBufferSize.EnvVar()))
+	})
+	tick := make(chan struct{})
+	fastComponent := &testSensorComponent{
+		t:          s.T(),
+		name:       "filtered",
+		tick:       tick,
+		responsesC: make(chan *message.ExpiringMessage, numberOfCentralMessages),
+	}
+	close(tick)
+
+	components := []common.SensorComponent{fastComponent}
+	s.receiver = NewCentralReceiver(s.finished, components...)
+
+	// Get initial metric values before test
+	initialDropped := getOpMetricValue(s.T(), fastComponent.Name(), "Drop")
+	initialAddOperations := getOpMetricValue(s.T(), fastComponent.Name(), "Add")
+	initialRemoveOperations := getOpMetricValue(s.T(), fastComponent.Name(), "Remove")
+
+	s.mockClient.EXPECT().Context().AnyTimes().Return(context.Background()).AnyTimes()
+
+	messagesFromCentral := make(chan *central.MsgToSensor, numberOfCentralMessages)
+	s.mockClient.EXPECT().Recv().MinTimes(5).DoAndReturn(func() (*central.MsgToSensor, error) {
+		msg, ok := <-messagesFromCentral
+		if !ok {
+			s.T().Logf("received EOF from central")
```

Comment:
```suggestion
			s.T().Log("received EOF from central")
```
I've seen this in other places, please make sure they are handled.